File: nodecore-cli/src/main/java/nodecore/cli/serialization/MultisigBundleInfo.java
// VeriBlock NodeCore
// Copyright 2017-2020 Xenios SEZC
// All rights reserved.
// https://www.veriblock.org
// Distributed under the MIT software license, see the accompanying
// file LICENSE or http://www.opensource.org/licenses/mit-license.php.

package nodecore.cli.serialization;

import nodecore.api.grpc.VeriBlockMessages;

public class MultisigBundleInfo {
    public MultisigBundleInfo(VeriBlockMessages.MultisigBundle multisigBundle) {
        multisigSlots = new MultisigSlotInfo[multisigBundle.getSlotsCount()];
        for (int i = 0; i < multisigBundle.getSlotsCount(); i++) {
            multisigSlots[i] = new MultisigSlotInfo(multisigBundle.getSlots(i));
        }
    }
    MultisigSlotInfo[] multisigSlots;
}

File: nodecore-cli/src/main/java/nodecore/cli/serialization/MultisigBundleInfo.kt
// VeriBlock NodeCore
// Copyright 2017-2020 Xenios SEZC
// All rights reserved.
// https://www.veriblock.org
// Distributed under the MIT software license, see the accompanying
// file LICENSE or http://www.opensource.org/licenses/mit-license.php.
package nodecore.cli.serialization

import nodecore.api.grpc.VeriBlockMessages.MultisigBundle

class MultisigBundleInfo(
    multisigBundle: MultisigBundle
) {
    val multisigSlots = Array(multisigBundle.slotsCount) {
        MultisigSlotInfo(multisigBundle.getSlots(it))
    }
}
